grpc/server/src/test/java/io/helidon/grpc/server/SslIT.java

```diff
@@ -0,0 +1,320 @@
+import services.EchoService;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+/**
+ * Tests for gRPC server with SSL connections
+ *
+ * @author Bin Chen
+ */
+public class SslIT {
+
+    // ----- data members ---------------------------------------------------
+
+    /**
+     * The {@link java.util.logging.Logger} to use for logging.
+     */
+    private static final Logger LOGGER = Logger.getLogger(SslIT.class.getName());
+
+    /**
+     * The Helidon {@link GrpcServer} being tested.
+     */
+    private static GrpcServer grpcServer_1WaySSL;
+
+    /**
+     * The Helidon {@link GrpcServer} being tested.
+     */
+    private static GrpcServer grpcServer_2WaySSL;
+
+    /**
+     * The Helidon {@link GrpcServer} being tested.
+     */
+    private static GrpcServer grpcServer_2WaySSLConfig;
+
+    /**
+     * Port used for 1waySSL
+     */
+    private static int m_nPort1WaySSL;
+
+    /**
+     * Port used for 2waySSL
+     */
+    private static int m_nPort2WaySSL;
+
+    /**
+     * Port used for 2waySSL using config-ssl.conf
+     */
+    private static int m_nPort2WaySSLConfig;
+
+    @Rule
+    public ExpectedException m_exception = ExpectedException.none();
+
+    private static final String CLIENT_CERT = "clientCert.pem";
+    private static final String CLIENT_KEY  = "clientKey.pem";
+    private static final String CA_CERT     = "ca.pem";
+    private static final String SERVER_CERT = "serverCert.pem";
+    private static final String SERVER_KEY  = "serverKey.pem";
+
+    private static String tlsCert;
+    private static String tlsKey;
+    private static String tlsCaCert;
+    private static String tlsClientKey;
+    private static String tlsClientCert;
+
+    private static String filePath;
+
+    // ----- test lifecycle -------------------------------------------------
+
+    @BeforeAll
+    public static void setup() throws Exception {
+        LogManager.getLogManager().readConfiguration(SslIT.class.getResourceAsStream("/logging.properties"));
+        File resourcesDirectory = new File("src/test/resources/ssl");
+        filePath = resourcesDirectory.getAbsolutePath();
+        tlsCert = getFile(SERVER_CERT);
+        tlsKey = getFile(SERVER_KEY);
+        tlsCaCert = getFile(CA_CERT);
+        tlsClientCert = getFile(CLIENT_CERT);
+        tlsClientKey = getFile(CLIENT_KEY);
+
+        AvailablePortIterator ports = LocalPlatform.get().getAvailablePorts();
+
+        m_nPort1WaySSL  = ports.next();
+        m_nPort2WaySSL  = ports.next();
+        m_nPort2WaySSLConfig = ports.next();
+
+        grpcServer_1WaySSL = startGrpcServer(m_nPort1WaySSL, false /*mutual*/, false /*useConfig*/);
+        grpcServer_2WaySSL = startGrpcServer(m_nPort2WaySSL, true /*mutual*/, false /*useConfig*/);
+        grpcServer_2WaySSLConfig = startGrpcServer(m_nPort2WaySSLConfig, true/*mutual*/, true /*useConfig*/);
+    }
+
+    @AfterClass
+    public static void cleanup() throws Exception
```

Comment:
Just occurred to me that we could call `shutdown` on each server in parallel, and then do `CompletableFuture.allOf(futures).get(10, TimeUnit.SECONDS);` at the end, instead of waiting for each server to shut down serially.

Not a big deal, so feel free to ignore the comment if you don't agree or understand what I'm asking, but it may speed up the test cleanup slightly and is a better way to handle async operations in general.

Reply:
Basically, the whole thing could be replaced by something like:
```
CompletableFuture<?>[] futures = 
    Set.of(grpcServer_1WaySSL, grpcServer_2WaySSL, grpcServer_2WaySSLConfig).stream()
        .filter(server -> server != null) // is any of them ever null??? do we even need this null check?
        .map(server -> server.shutdown().toCompletableFuture())
        .toArray();

CompletableFuture.allOf(futures).get(10, TimeUnit.SECONDS);
```

Reply:
I made the change, but a little different as Java8 does not support Set.of.